src/book/entities/priceOps.ts

```diff
@@ -0,0 +1,31 @@
+import type { DateTime } from 'luxon';
+import type { PriceDBMap } from '../prices';
+import type Commodity from './Commodity';
+import Price from './Price';
+
+type GetExchangeRateOptions = {
+  date: DateTime;
+  from: Commodity;
+  to: Commodity;
+};
+
+const getExchangeRate = (prices: PriceDBMap, options: GetExchangeRateOptions): Price => {
+  const { from, to, date } = options;
+
+  const isSameCommodity = from.guid === to.guid;
+  if (isSameCommodity) {
+    return Price.create({ valueNum: 1, valueDenom: 1 });
+  }
+
+  if (to.namespace !== 'CURRENCY') {
+    return Price.create({ valueNum: 1, valueDenom: 1 });
+  }
+
+  return from.namespace !== 'CURRENCY'
+    ? prices.getInvestmentPrice(from.mnemonic, date)
+    : prices.getPrice(from.mnemonic, to.mnemonic, date);
+};
+
+export const priceOps = {
+  getExchangeRate,
+};
```

Comment:
I'm thinking that maybe this one could be a method of the `PriceDBMap`. I've found myself repeating this logic in other places around the code where you have to check if it's investment or price or return a 1 so.. happy to move it there.